hypertension/dc-7101-algorithms/py-root/lib/main.py

```diff
@@ -1,12 +1,59 @@
 from typing import Dict
+from .algorithms.bp_sufficiency import sufficient_to_autopopulate
+from .algorithms.bp_history import history_of_diastolic_bp
+from .algorithms.continuous_medication import continuous_medication_required
+from .algorithms.utils import hypertension_medications
 
 def main(event: Dict):
+    """
+    Take a request that includes hypertension related data, and return a response
+
+    :param event: request body
+    :type event: dict
+    :return: response body
+    :rtype: dict
+    """
+
+    statusCode = 200
+    try:
```

Comment:
if the data is in an incorrect format, we'll get an error, and (afaik) in order to return a 500 error (since we're not using a framework that handles it for us - though maybe lamda does? I haven't been able to test it yet) I would need to catch that error and return it. Upon writing this, it could be that the lambda function does that automatically, idk

Reply:
I guess it [does](https://docs.aws.amazon.com/lambda/latest/dg/python-exceptions.html)? if so I can remove this try catch block

Reply:
Ah.

Generally speaking, try/catch (try/except) should only be used to handle errors that are out of control of the code (and the programmer who wrote it). For example, we wrap code that attempts to connect to a database in try/catch because, at run time, the programmer and code have zero control as to whether that operation is going to be successful (e.g. that database might be down when that code runs).

However, for things that are under the control of the programmer/code, such as validating the data passed in, we want to gracefully handle things things. One could argue that try/catch is a graceful way to handle it. IMO try/catch should reserved as much as possible out of the control of the programmer, and other language features such as `if` statements should be used to gracefully handling things under the control of the programmer and their code.

(Generally, I got this philosophy from https://www.joyent.com/node-js/production/design/errors).

I just created ticket https://amida.atlassian.net/browse/MCP-1047 to refactor this code to eliminate the try/catch in favor of Swagger API specs as a part of the AWS API Gateway, and/or other Python language features (such as `if` statements and/or libraries dedicated to data validation).